mm2src/coins/z_coin.rs

```diff
@@ -323,22 +325,21 @@ impl ZCoin {
         })
     }
 
-    #[cfg(not(target_arch = "wasm32"))]
-    async fn my_balance_sat(&self) -> Result<u64, MmError<ZcashClientError>> {
+    async fn my_balance_sat(&self) -> Result<u64, MmError<ZCoinBalanceError>> {
         let wallet_db = self.z_fields.light_wallet_db.clone();
-        async_blocking(move || {
+        #[cfg(target_arch = "wasm32")]
+        let balance_sat: u64 = wallet_db.db.get_balance(AccountId::default()).await?.into();
+
+        #[cfg(not(target_arch = "wasm32"))]
+        let balance_sat: u64 = async_blocking(move || {
             let db_guard = wallet_db.db.inner();
             let db_guard = db_guard.lock().unwrap();
-            let balance = get_balance(&db_guard, AccountId::default())?.into();
-            Ok(balance)
+            get_balance(&db_guard, AccountId::default()).map(|v| v.into())
         })
         .await
-    }
+        .map_to_mm(|err| ZCoinBalanceError::BalanceError(err.to_string()))?;
 
-    #[cfg(target_arch = "wasm32")]
-    async fn my_balance_sat(&self) -> Result<u64, MmError<ZCoinBalanceError>> {
-        let wallet_db = self.z_fields.light_wallet_db.clone();
-        Ok(wallet_db.db.get_balance(AccountId::default()).await?.into())
+        Ok(balance_sat)
```

Comment:
This looks iffy to me. I liked the previous separation.

Reply:
I mean not really..

```rust

    async fn my_balance_sat(&self) -> Result<u64, MmError<ZCoinBalanceError>> {
        let wallet_db = self.z_fields.light_wallet_db.clone();

        #[cfg(target_arch = "wasm32")]
        let balance_sat: u64 = wallet_db.db.get_balance(AccountId::default()).await?.into();

        #[cfg(not(target_arch = "wasm32"))]
        let balance_sat: u64 = async_blocking(move || {
            let db_guard = wallet_db.db.inner();
            let db_guard = db_guard.lock().unwrap();
            get_balance(&db_guard, AccountId::default()).map(|v| v.into())
        })
        .await
        .map_to_mm(|err| ZCoinBalanceError::BalanceError(err.to_string()))?;

        Ok(balance_sat)
    }


```